### PR TITLE
Bugfix/678/1.5.0 encoding normalization zos copy

### PR DESCRIPTION
**Pull request by rexemin**
##### SUMMARY
Fixes both #678 and #664 for 1.5.0, since both have to do with encoding issues in zos_copy.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
zos_copy

##### ADDITIONAL INFORMATION
These issues deal with problems we've seen when users are trying to copy files that are not encoded in UTF-8 or IBM-1047, and even worse, when those files may have CRLF endings.

I added tests to verify that encoding normalization works for the following two cases:
1. When trying to compute the record length for a dataset.
2. When trying to get rid of newlines from files with CRLF endings.

### CODE REVIEW

**Comment by rexemin** [comment]
Jenkins pipeline
Screenshot 2023-04-04 at 16 16 02
Screenshot 2023-04-04 at 16 16 19

**Review by richp405 — APPROVED** [review]
looks okay to me.  I was surprised about IBM037 (cp 37)... used to hearing about 1047 instead, but I looked it up, and you are correct.

File: plugins/modules/zos_copy.py [hunk 1]
```diff
@@ -1454,7 +1465,7 @@ def get_file_record_length(file):
     """
     max_line_length = 0
 
-    with open(file, "r") as src_file:
+    with open(file, "r", encoding="utf-8") as src_file:
```
Comment (ddimatos) [inline plugins/modules/zos_copy.py:1468]: I have a question, so `get_file_record_length` is used to determine record length of a file should we enter into that use case. My question is you open in utf-8 always but if the source is 1047, python would try to read its utf-8. I thought opening a file with encoding is just overriding the default behavior which detects the systems locale but the file's actual encoding has to match what you pass as encoding.

In other words, can you always use utf-8 when the source could be another encoding or am I missing something in the code.

Comment (ddimatos) [reply plugins/modules/zos_copy.py:1468]: I might have figured it out, looks like you normalize on this block of code:
```
                    os.close(fd)
                    encode_utils.uss_convert_encoding(
                        new_src,
                        converted_src,
                        src_tag,
                        "UTF-8"
                    )
```

**Review by ddimatos — APPROVED** [review]
I am approving this @rexemin, nice work. I do have one question that I may have figured out but could use some clarity on. Thank you .

File: plugins/modules/zos_copy.py (hunk 1 above)
Comment (rexemin) [reply plugins/modules/zos_copy.py:1468]: Yes, before getting to `get_file_record_length` the file gets normalized.
